### ISSUE TITLE
Life360 Server Error Logging In:

### ISSUE THREAD

**Issue by ctreischman** [hepcat72/node-red-contrib-life360#18]
Yesterday I started receiving this:

"Life360 server error logging in: 403 Error: Request failed with status code 403 Client is not valid/enabled"

I've logged out and into the app successfully on my phone.  This began in the middle of the day yesterday so nothing else should have changed. 

Chris

**Comment by hepcat72** [comment]
Sheesh. Yeah, I have 403's since yesterday as well, but my response is:
```
22 Aug 15:13:13 - [error] [Life360-Server:Life360-Server] Life360 server error getting circles: 403 Error: Request failed with status code 403 undefined
```

**Comment by hepcat72** [comment]
Restarted node-red, now I see the client is not valid/enabled error.

**Comment by hepcat72** [comment]
Again, people smarter than me already [fixed it elsewhere](https://community.hubitat.com/t/life360-broken/122202/29). I now have my instance running. I will push an update shortly.

**Comment by hepcat72** [comment]
They changed the port on which it runs and this time changing the client secret was necessary.

**Comment by hepcat72** [comment]
I now feel like they're actively working on the API, so I somewhat expect this to keep happening as they work on their code...

**Comment by hepcat72** [comment]
Before I push out a new version, you can make the changes yourself.  Change:
```
const LIFE360_API = "https://api-cloudfront.life360.com/v3"
```
to:
```
const LIFE360_API = "https://api-cloudfront.life360.com:443/v3"
```
and change:
```
const LIFE360_CLIENT_SECRET = "cFJFcXVnYWJSZXRyZTRFc3RldGhlcnVmcmVQdW1hbUV4dWNyRUh1YzptM2ZydXBSZXRSZXN3ZXJFQ2hBUHJFOTZxYWtFZHI0Vg==";
```
to:
```
const LIFE360_CLIENT_SECRET = "Y2F0aGFwYWNyQVBoZUtVc3RlOGV2ZXZldnVjSGFmZVRydVl1ZnJhYzpkOEM5ZVlVdkE2dUZ1YnJ1SmVnZXRyZVZ1dFJlQ1JVWQ==";
```

**Comment by ctreischman** [comment]
I made the change.  What is the best way to restart nodered without
restarting my computer?

**Comment by ctreischman** [comment]
Success!  The changes worked.  Thanks again for your support.

**Comment by hepcat72** [comment]
Good. But since you asked - it depends on how you set up node red.  I use pm2 to manage the node-red process, so for me, restarting node-red is `pm2 restart node-red`.

**Comment by hepcat72** [comment]
OK, the update has been published.  Version bump to `1.1.2`.  Might take 5 minutes or so before it shows up in the palette manager.

**Comment by StoricU** [comment]
Anyone else got the problem again?
25.9.2023, 10:37:42[node: Life360 - public@](http://192.168.10.16:1880/#)
msg : string[124]
"Life360 server error logging in: 403 Error: Request failed with status code 403 Invalid login and password. Please try again"

Have not changed anything and the login and password is working in the app.

**Comment by hepcat72** [comment]
I am not currently. Did you change your password? There used to be a problem with unsupported special characters. It's in one of the closed issues I think.

**Comment by hepcat72** [comment]
I wonder if Life360 is having issues. I didn't get either of the notifications I am supposed to get about my wife going to work. The app shows where she is though.

**Comment by hepcat72** [comment]
~Never mind. My wife forgot her phone, lol.~

Nope... the map showed her at home and I assumed she forgot her phone. It just updated maybe a half hour later. Though for all I know, she just had her phone off or something.

**Comment by StoricU** [comment]
Strange, since the username and password have been working for several weeks without any issues.. Except the issue that got fixed in 1.1.2.
I'll try to change the password later today just to be sure.

**Comment by hepcat72** [comment]
My initial error was different from the errors I got after a restart.  It wasn't invalid password. Mine was something like "invalid client" or something. It just could be that the presentation of the issue before 1.1.2 is variable... Glad it's fixed for you. No need to change your password if you got it running.
